src/player.cpp

```diff
@@ -11009,6 +11013,57 @@ void player::burn_move_stamina( int moves )
     }
 }
 
+void player::mount_creature( monster &z )
+{
+    tripoint pnt = z.pos();
+    auto mons = g->critter_tracker->find( pnt );
+    if( mons == nullptr ) {
+        add_msg( m_debug, "mount_creature() : monster not found in critter_tracker" );
+        return;
+    }
+    add_effect( effect_riding, 1_turns, num_bp, true );
+    z.add_effect( effect_ridden, 1_turns, num_bp, true );
+    if( z.has_effect( effect_tied ) ) {
+        z.remove_effect( effect_tied );
+        if( z.tied_item ) {
+            i_add( *z.tied_item );
+            z.tied_item = cata::nullopt;
+        }
+    }
+    z.mounted_player_id = getID();
+    if( z.has_effect( effect_harnessed ) ) {
+        z.remove_effect( effect_harnessed );
+        add_msg_if_player( m_info, _( "You remove the %s's harness." ), z.get_name() );
+    }
+    mounted_creature = mons;
+    mons->mounted_player = this;
+    if( is_hauling() ) {
+        stop_hauling();
+    }
+    if( !is_npc() ) {
```

Comment:
```suggestion
    if( is_player() ) {
```
And change the `add_msg_if_player` to just `add_msg` so the block is consistent with the same block in ` player::forced_dismount`. Or just move the whole block into another function and call it from here and from  ` player::forced_dismount`.